jinahub/indexers/query/vector/FaissIndexer/__init__.py

```diff
@@ -1,69 +1,72 @@
     """
 
-    def __init__(self,
-                 index_key: str,
-                 train_filepath: Optional[str] = None,
-                 max_num_training_points: Optional[int] = None,
-                 requires_training: bool = True,
-                 distance: str = 'l2',
-                 normalize: bool = False,
-                 nprobe: int = 1,
-                 *args,
-                 **kwargs):
-        """
-        Initialize an Faiss Indexer
-
-        :param index_key: index type supported by ``faiss.index_factory``
-        :param train_filepath: the training data file path, e.g ``faiss.tgz`` or `faiss.npy`. The data file is expected
-            to be either `.npy` file from `numpy.save()` or a `.tgz` file from `NumpyIndexer`. If none is provided, `indexed` data will be used
-            to train the Indexer (In that case, one must be careful when sharding is enabled, because every shard will be trained with its own part of data).
-            The data will only be loaded if `requires_training` is set to True.
-        :param max_num_training_points: Optional argument to consider only a subset of training points to training data from `train_filepath`.
-            The points will be selected randomly from the available points
-        :param requires_training: Boolean flag indicating if the index type requires training to be run before building index.
-        :param distance: 'l2' or 'inner_product' accepted. Determines which distances to optimize by FAISS. l2...smaller is better, inner_product...larger is better
-        :param normalize: whether or not to normalize the vectors e.g. for the cosine similarity https://github.com/facebookresearch/faiss/wiki/MetricType-and-distances#how-can-i-index-vectors-for-cosine-similarity
-        :param nprobe: Number of clusters to consider at search time.
-
-        .. highlight:: python
-        .. code-block:: python
-            # generate a training file in `.tgz`
-            import gzip
-            import numpy as np
-            from jina.executors.indexers.vector.faiss import FaissIndexer
-
-            train_filepath = 'faiss_train.tgz'
-            train_data = np.random.rand(10000, 128)
-            with gzip.open(train_filepath, 'wb', compresslevel=1) as f:
-                f.write(train_data.astype('float32'))
-            indexer = FaissIndexer('PCA64,FLAT', train_filepath)
-
-            # generate a training file in `.npy`
-            train_filepath = 'faiss_train'
-            np.save(train_filepath, train_data)
-            indexer = FaissIndexer('PCA64,FLAT', train_filepath)
-        """
+    def __init__(
+        self,
+        index_key: str,
+        train_filepath: Optional[str] = None,
+        max_num_training_points: Optional[int] = None,
+        requires_training: bool = True,
+        distance: str = 'l2',
+        normalize: bool = False,
+        nprobe: int = 1,
+        dump_path: Optional[str] = None,
+        traverse_path: str = 'r',
```

Comment:
we have to agree if this is `default_traversal_path` or what, but we should be consistent. Also this should be a `Union[str, List[str]]`

Reply:
That's why we should actually have another class, `Indexer` to do this consistently. 

I'll name it `default_` for this